### PR TITLE
Fix Issue 21092 -[ICE] Segmentation fault in ExpressionPrettyPrintVis…

### PR DESCRIPTION
**Pull request by RazvanN7**
…itor::visit(CommaExp*) at dmd/hdrgen.d:2293

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]
Please target stable.

**Comment by RazvanN7** [comment]
The code that this PR fixes has not yet made it to stable.

File: src/dmd/hdrgen.d [hunk 1]
```diff
@@ -2276,7 +2276,7 @@ public:
 
         // not a CommaExp introduced for temporaries, go on
         // the old path
-        if (!ve || !ve.var.storage_class & STC.temp)
+        if (!ve || !(ve.var.storage_class & STC.temp))
```
Comment (RazvanN7) [inline src/dmd/hdrgen.d:2279]: This is really embarrassing....

Comment (MoonlightSentinel) [reply src/dmd/hdrgen.d:2279]: Good example for #11077

**Comment by UplinkCoder** [comment]
I'll try the fix now

**Comment by UplinkCoder** [comment]
@RazvanN7 please disable the omitting of the temporary when `fullDump` i.e. `-vcg-ast` is given

**Comment by UplinkCoder** [comment]
Yes this fixes it.
Should _really_ have targeted stable.

**Comment by Geod24** [comment]
https://github.com/dlang/dmd/pull/11500#issuecomment-668357904

**Comment by UplinkCoder** [comment]
@Geod24 I am aware
@RazvanN7 Good work. Thanks for the quick fix.

**Review by UplinkCoder — COMMENTED** [review]
LGTM

**Comment by kinke** [comment]
This should have gone into stable indeed: #11548

**Pull request merged**